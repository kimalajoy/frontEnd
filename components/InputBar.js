import React, {Component, useState} from 'react';
import { View, Text, Button, TextInput } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

import styles from '../styles/styles';

const searchIcon = <Ionicons name="md-search" size={40} color="black" />

class InputBar extends Component {
  
  render () {
    
    return (
      <View style={styles.searchInputContainer}>
        <TextInput
          style={styles.searchbox}
            placeholder={this.props.placeholder}
            onChangeText={(text) => this.props.setValue(text)}
        />
        <View style={styles.searchButton}>
          <Button
            title={this.props.buttonText}
            onPress={() => this.props.inputButtonClick()}
          />
        </View>
      </View>
    )
  }
}

export default InputBar;
