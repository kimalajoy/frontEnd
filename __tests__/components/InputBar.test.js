import React from 'react';
import { render, cleanup } from 'react-native-testing-library';

import InputBar from '../../components/InputBar';

describe("Landing Page / Home Screen", () => {
  let inputBar;

  beforeEach(() => {
    const onPressMock = jest.fn()
    const onChangeMock = jest.fn()

    inputBar = render(<InputBar
      placeholder={'Test Placeholder Text'}
      buttonText={'Test Button Text'}
      setValue={onChangeMock}
      inputButtonClick={onPressMock}
      />)
  });

  afterEach(cleanup);

  test('test', () => {
    expect(true).toBeTruthy()
  })
})
