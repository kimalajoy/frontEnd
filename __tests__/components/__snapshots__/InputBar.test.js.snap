// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page / Home Screen it renders snapshot test 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Test Placeholder Text"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 254, 254, .1)",
          "borderColor": "#000",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 40,
          "padding": 5,
          "width": "100%",
        },
        Object {
          "color": "#1d1d1d",
          "fontSize": 19,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "borderRadius": 5,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#5DA25B",
          "borderRadius": 5,
          "justifyContent": "flex-end",
          "marginTop": 5,
          "opacity": 1,
          "padding": 8,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fafafa",
            "fontSize": 19,
          }
        }
      >
        Test Button Text
      </Text>
    </View>
  </View>
</View>
`;
